src/mlpack/methods/ann/augmented/tasks/add.hpp

```diff
@@ -26,8 +26,9 @@ namespace tasks /* Task utilities for augmented */ {
  * The parameters are:
  * - macimum binary length;
  * 
- * Every element of sequence is encoded as 3-dimensional vector
- * (one-hot vector for elements {0, 1, +}).
+ * Every element of sequence is encoded as 1-dimensional vector
+ * (possible vector elements are {0, 1, 0.5} -
+ * the latter corresponds to '+' sign').
```

Comment:
I guess, I don't mean to get too stuck on this one point, but when I look at the HAM paper, their results show about 46% error rate for the LSTM network, and I think the input length is 5 bits in that case.  (It doesn't actually say when the `Add` task is introduced, but the other tasks around it are 5 bits, so that is my guess.)  But our 5-bit results are much worse, as seen in Marcus' results in the models#1 PR:

```
➜  bin git:(master) ✗ ./run_add_task.sh 
Evaluating bit_length 1
Final score: 0.460938
Evaluating bit_length 2
Final score: 0.28125
Evaluating bit_length 3
Final score: 0.109375
Evaluating bit_length 4
Final score: 0.0390625
Evaluating bit_length 5
Final score: 0.0234375
Evaluating bit_length 6
Final score: 0.0234375
Evaluating bit_length 7
Final score: 0
Evaluating bit_length 8
Final score: 0.0078125
Evaluating bit_length 9
Final score: 0
Evaluating bit_length 10
Final score: 0
```

Do you have up-to-date results for the 3-dimensional input scheme that they are using in that paper?  I see that you and @zoq discussed it at length earlier in the PR, but I think at that point, there may have been some other bugs in the implementation.  So if you try with the three-dimensional inputs like you had it originally, does it give any performance improvement?  

I could very easily be wrong, but I think that encoding all of this into one dimension makes the task more difficult, because the network must also learn that 0, 1, and 0.5 mean completely different things; in 3 dimensions, that information is already "split".

In addition, I think the "final output symbol" is not included in this implementation; is that correct?  That may also help improve performance.